app/src/main/java/protect/card_locker/ManageGroupsActivity.kt

```diff
@@ -1,247 +1,240 @@
+            dialog.cancel()
+        }
+        val dialog = builder.create()
 
         // Now that the dialog exists, we can bind something that affects the OK button
-        input.addTextChangedListener(new SimpleTextWatcher() {
-            public void onTextChanged(CharSequence s, int start, int before, int count) {
-                String groupName = s.toString().trim();
-
-                if (groupName.length() == 0) {
-                    input.setError(getString(R.string.group_name_is_empty));
-                    dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false);
-                    return;
-                }
-
-                if (DBHelper.getGroup(mDatabase, groupName) != null) {
-                    input.setError(getString(R.string.group_name_already_in_use));
-                    dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false);
-                    return;
-                }
-
-                input.setError(null);
-                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(true);
+        input.doOnTextChanged { s: CharSequence?, start: Int, before: Int, count: Int ->
+            val groupName = s?.trim().toString()
+
+            if (groupName.isEmpty()) {
+                input.error = getString(R.string.group_name_is_empty)
+                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false)
+                return@doOnTextChanged
             }
-        });
 
-        dialog.show();
+            if (DBHelper.getGroup(mDatabase, groupName) != null) {
+                input.error = getString(R.string.group_name_already_in_use)
+                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false)
+                return@doOnTextChanged
+            }
+
+            input.error = null
+            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(true)
+        }
+
+        dialog.apply {
+            show()
+            // Disable button (must be done **after** dialog is shown to prevent crash
+            getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false)
+            // Set focus on input field
+            window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE)
+        }
 
-        // Disable button (must be done **after** dialog is shown to prevent crash
-        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false);
-        // Set focus on input field
-        dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
-        input.requestFocus();
+        input.requestFocus()
     }
 
-    private String getGroupName(View view) {
-        TextView groupNameTextView = view.findViewById(R.id.name);
-        return (String) groupNameTextView.getText();
+    private fun getGroupName(view: View): String? {
+        val groupNameTextView = view.findViewById<TextView>(R.id.name)
+        return groupNameTextView.text as String?
     }
```

Comment:
If I look at the Android docs, a TextView should always return a CharSequence. So I'd think this shouldn't be possible to be null?

https://developer.android.com/reference/android/widget/TextView#getText()